packages/venia-ui/lib/components/Gallery/items.js

```diff
@@ -34,27 +34,35 @@ class GalleryItems extends Component {
             : defaultPlaceholders;
     }
 
-    // map Magento 2.3.1 schema changes to Venia 2.0.0 proptype shape to maintain backwards compatibility
-    mapGalleryItem(item) {
-        const { small_image } = item;
-        return {
-            ...item,
-            small_image:
-                typeof small_image === 'object' ? small_image.url : small_image
-        };
-    }
-
     render() {
         const { items } = this.props;
 
         if (items === emptyData) {
             return this.placeholders;
         }
 
-        return items.map(item => (
-            <GalleryItem key={item.id} item={this.mapGalleryItem(item)} />
-        ));
+        return items.map(item => <GalleryItem key={item.id} item={item} />);
     }
 }
+const PureGalleryItems = memo(GalleryItems);
```

Comment:
Instead of doing this can you convert `GalleryItems` into a function component and utilize `useMemo` within? The problem with our original code is that we are calling `mapGalleryItem` _every_ time when we really should only call it when `items` change, as you've done in the container. I think we can accomplish the same thing without the wrapping container.

It could even be useful to memoize the `placeholders` function :)

Reply:
After a closer look the meoized map would probably fit better in [category.js](https://github.com/amsterdamharu/pwa-studio/blob/d08891f/packages/venia-ui/lib/RootComponents/Category/category.js#L46) as that is where the data is fetched. Other components just seem to pass it along. Maybe close this pull request, if you want to memoize mapping the galery item data then I can create a new one doing the mapping in [category.js](https://github.com/amsterdamharu/pwa-studio/blob/d08891f/packages/venia-ui/lib/RootComponents/Category/category.js)


I see you may do a [complete overhaul](https://github.com/magento/pwa-studio/tree/peregrine-implementation/convert-to-function-components/packages/venia-ui) to the venia-ui so this will resolve itself if/when you merge that branch. Please let me know if you need help with that. 

On another topic; at the moment filter is written to state.catalog.chosenFilterOptions and to the url but not picked up in the query. Will that be implemented in [catory.js](https://github.com/amsterdamharu/pwa-studio/blob/d08891f/packages/venia-ui/lib/RootComponents/Category/category.js#L50?) as well?

Reply:
> After a closer look the meoized map would probably fit better in category.js as that is where the data is fetched. Other components just seem to pass it along. Maybe close this pull request, if you want to memoize mapping the galery item data then I can create a new one doing the mapping in category.js

What you're talking about is normalizing the data at some point. I agree we should probably raise it to the highest point that it makes sense. In this case we aren't saving the fetched data in the redux store as far as I can tell so doing this in the `category` component would be OK.

> I see you may do a complete overhaul to the venia-ui so this will resolve itself if/when you merge that branch. Please let me know if you need help with that.

I'll be trying to get this wrapped up this week, but no promises ;)

On another topic; at the moment filter is written to state.catalog.chosenFilterOptions and to the url but not picked up in the query. Will that be implemented in catory.js as well?

Filters currently are not functional in the graphql API. We've got the UI hooked up but we don't ever actually make a request to filter right now. There's an issue somewhere in here for filtering that you can follow.